dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,682 @@
+
+                    // SAML is configurated
+                    final boolean isLogoutNeed = this.samlConfigurationService.getConfigAsBoolean(
+                            identityProviderConfiguration, SamlName.DOTCMS_SAML_IS_LOGOUT_NEED);
+
+                    // check if there is any exception filter path, to avoid to
+                    // canApply all the logic.
+                    if (!this.checkAccessFilters(request.getRequestURI(),
+                            this.getAccessFilterArray(identityProviderConfiguration))
+                            && this.checkIncludePath(request.getRequestURI(),
+                            this.getIncludePathArray(identityProviderConfiguration), request)) {
+
+                        final AutoLoginResult autoLoginResult = this.autoLogin(request, response,
+                                session, identityProviderConfiguration);
+
+                        // we have to assign again the session, since the doAutoLogin might be renewed.
+                        session = autoLoginResult.getSession();
+
+                        // if the auto login couldn't logged the user, then send it
+                        // to the IdP login page (if it is not already logged in).
+                        if (null == session || this.isNotLogged(request, session)) {
+
+                            Logger.debug(this, ()->"There's no logged-in user. Processing SAML request...");
+                            this.doRequestLoginSecurityLog(request, identityProviderConfiguration);
+
+                            final String originalRequest = request.getRequestURI() +
+                                    null != request.getQueryString()?
+                                        "?" + request.getQueryString() : StringUtils.EMPTY;
+
+                            redirectAfterLogin = UtilMethods.isSet(request.getParameter(REFERRER_PARAMETER_KEY))
+                                    ? request.getParameter(REFERRER_PARAMETER_KEY) :
+                                    // this is safe, just to make a redirection when the user get's logged.
+                                    originalRequest;
+
+                            Logger.debug(this.getClass(),
+                                    "Executing SAML Login Redirection with request: " + redirectAfterLogin);
+
+                            // if we don't have a redirect yet
+                            if (null != session) {
+
+                                session.setAttribute(WebKeys.REDIRECT_AFTER_LOGIN, redirectAfterLogin);
+                                session.setAttribute(ORIGINAL_REQUEST,             originalRequest);
+                            }
+
+                            try {
+                                // this will redirect the user to the IdP Login Page.
+                                this.samlAuthenticationService.authentication(request, response, identityProviderConfiguration);
+                            } catch (Exception exception) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)